source/rules/no-exposed-subjects.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * @license Use of this source code is governed by an MIT-style license that
+ * can be found in the LICENSE file at https://github.com/cartant/eslint-plugin-rxjs
+ */
+
+import { Rule } from "eslint";
+import * as es from "estree";
+import { couldBeType } from "tsutils-etc";
+import { getTypeCheckerAndNodeMap } from "../utils";
+
+const defaultAllowedTypesRegExp = /^EventEmitter$/;
+
+const rule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      category: "RxJS",
+      description: "Disallows exposed subjects.",
+      recommended: false
+    },
+    fixable: null,
+    messages: {
+      forbidden: "Subject '{{subject}}' must be private.",
+      forbiddenAllowProtected:
+        "Subject '{{subject}}' must be private or protected."
+    },
+    schema: [
+      {
+        properties: {
+          allowProtected: { type: "boolean" }
+        },
+        type: "object"
+      }
+    ]
+  },
+  create: context => {
+    const [config] = context.options;
+    const { allowProtected = false } = config || {};
+    const { nodeMap, typeChecker } = getTypeCheckerAndNodeMap(context);
+
+    const messageId = allowProtected ? "forbiddenAllowProtected" : "forbidden";
+    const accessibilityRexExp = allowProtected
+      ? /^(private|protected)$/
+      : /^private$/;
+
+    function isSubject(node: es.Node) {
+      const tsNode = nodeMap.get(node);
+      const tsType = typeChecker.getTypeAtLocation(tsNode);
+
+      return (
+        couldBeType(tsType, "Subject") &&
+        !couldBeType(tsType, defaultAllowedTypesRegExp)
+      );
+    }
+
+    return {
+      [`ClassProperty[accessibility!=${accessibilityRexExp}]`]: (
+        node: es.Node // there is no `es.ClassProperty` type
```

Comment:
Is there a specific type I could use here instead of `es.Node`?
A better type would also resolve the need to cast it as `any`...

Reply:
IDK. The `typescript-eslint` peeps told me that the ESLint types were lacking, so I'm not surprised. What you've done is fine.